treeherder/model/models.py

```diff
@@ -187,17 +191,21 @@ class Commit(models.Model):
     revision = models.CharField(max_length=40, db_index=True)
     author = models.CharField(max_length=150)
     comments = models.TextField()
+    search_vector = SearchVectorField(null=True, blank=True)
 
     class Meta:
         db_table = "commit"
         unique_together = ("push", "revision")
         indexes = [
-            GinIndex(
-                SearchVector("revision", "author", Substr("comments", 1, 100000), config="english"),
-                name="search_vector_idx",
-            ),
+            GinIndex(fields=["search_vector"], name="search_vector_idx"),
         ]
 
+    def save(self, *args, **kwargs):
+        self.search_vector = SearchVector(
+            self.revision, self.author, Substr(self.comments, 1, 100000), config="english"
```

Comment:
I have a couple of questions:
1. would it be possible to add the search_vector column to the push model instead of the commit model? This would save us a join.
2. I believe we can reduce the value `100000`, we are really interested in the first line. Actually I wonder if we could just get the first line (that is, substr to the first "\n")?

Reply:
I think we would still need a join to access `comments` if we move up to the push model